new-cep.md

```diff
@@ -0,0 +1,173 @@
+        run_for={"install", "create"}
+    )
+
+
+def custom_plugin_post_run_action():
+    """
+    Defines our custom post-run action which simply prints a message.
+    """
+    print("post-run action")
+
+
+@hookimpl
+def conda_post_run():
+    """
+    Returns our CondaPreRun class which attaches our ``custom_plugin_post_run_action`` to
+    the "install" and "create" command.
+    """
+    yield CondaPreRun(
+        name=f"{PLUGIN_NAME}_post_run",
+        action=custom_plugin_post_run_action,
+        run_for={"install", "create"}
+    )
+```
+
+### `on_exception`
+
+For the `on_exception` hook, plugin authors will begin by defining a `conda.plugins.hookimpl` decorated 
+function called `conda_on_exception`. This function will return the `CondaOnException` class with the
+following attributes:
+
+- `name`: unique name which identifies this plugin hook
+- `action`: a callable which contains the code to be run
+
+#### Example
+
+```python
+from conda.plugins import hookimpl, CondaOnException
+
+
+PLUGIN_NAME = "custom_plugin"
+
+
+def custom_plugin_on_exception_action():
+    """
+    Defines our custom on_exception action which simply prints a message.
+    """
+    print("on_exception action")
+
+
+@hookimpl
+def conda_on_exception():
+    """
+    Returns our CondaOnException class which attaches our ``custom_plugin_on_exception_action``.
+    """
+    yield CondaPreRun(
+        name=f"{PLUGIN_NAME}_on_exception",
+        action=custom_plugin_on_exception_action
+    )
+```
+
+
+## Motivation
+
+Our immediate motivations for adding these plugin hooks include the following:
+
+### Better authentication handling
+
+The `pre_run` hook will enable plugin authors to interrupt the normal start up of conda commands.
+For better authentication handling, this opens the door for either asking the user directly
+for their credentials or retrieving credentials from an OS keyring. These credentials can
+then be stored and used for the duration of the running command.
+
+A draft of this could function can be found in the following pull request:
+
+- https://github.com/conda/conda/pull/12139
+
+### Better exception logging
+
+The `on_exception` hook will enable plugin authors to use custom logging mechanisms for exceptions.
+One use case could be hooking up conda to the popular Sentry error logging platform (see: https://sentry.io/welcome/).
+
+### Plus more
+
+The above use cases are just a few possibilities out there. We know that our users and community will
+have many more ideas for how best to utilize these generic plugin hooks.
+
+## Rationale
+
+By introducing a set of generic hooks like the above, we grant plugin authors with quite a bit of
+flexibility for customizing how conda behaves. We believe starting out with a small set of generic
+plugin hooks is the best so we do not overwhelm would be developers at first. Depending on how much
+these are used and whether there is a demand, we may choose add more in the future with a new CEP.
```

Comment:
```suggestion
plugin hooks is the best so that we do not overwhelm potential conda contributors. Depending on how
much these are used and whether there is a demand, we may choose to add more in the future with a new CEP.
```